### PR TITLE
Recipe - ror capstone project

### PR DESCRIPTION
**Pull request by sylvester-wamaya**
The following were added and implemented:
### Login page and registration page:
- Built with Devise.

### Food list
- Display a list of food added by the logged-in user, also display the quantity of a given food.
- Lead to a form that allows users to add new food.

### Recipes list:
- Display a list of recipes created by the logged-in user as in the wireframe.
- Lead to recipe details.
- If the user is the owner of the recipe, allow the user to delete it.

### Public recipe list:
- Display a list of all public recipes.
- Lead to recipe details.

### Recipe details:
- Display a toggle button that allows for a recipe to be made public or private.
- If the recipe is public or the user is the owner of the recipe, Display the recipe details as in the wireframe.
- If the user is the owner of the recipe, lead to the form that allows the user to add new food.

### General shopping list view : 
- Show the list of food that is missing for all recipes of the logged-in user.
- Count the total food items and total price of the missing food.

- No N+1 queries happening.

- Created a navigation menu that allows users to open all of the pages you created.
- Wrote unit and integration tests

### CODE REVIEW

File: app/views/recipes/public_recipes.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<section class="public">
+    <h1>Public Recipes</h1>
+    <ul>
+        <% @public_recipes.includes([:recipe_foods]).each do |public_recipe| %>
+        <li>
+        <%= render partial: "public_recipe", locals: {recipe: public_recipe}%>
+        </li>
+        <% end %>
+    </ul>
+
```
Comment (thecodechaser) [inline app/views/recipes/public_recipes.html.erb:10]: - [ ] Please make sure to display recipes ordered by the newest on the public recipes page as requested in the project requirements. Currently, recipes are displayed ordered by the oldest.

image

File: app/views/shopping_list/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        <td><%= link_to food.name, food_path(food), class: 'food-name' %></td>
+        <td>$<%= food.price * food.quantity %></td>
+      </tr>
+    <% end %>
+  </tbody>
+  <tfoot>
+    <tr>
+      <td><strong>Total</strong></td>
+      <td><strong>$<%= @unused_foods.sum { |food| food.price * food.quantity } %></strong></td>
+    </tr>
```
Comment (thecodechaser) [inline app/views/shopping_list/index.html.erb:22]: - [ ] In the General shopping list you should display the food that is missing from all recipes. (Example: You have a food apple with 20 quantities and in a recipe, you added food ingredient apple in 30 quantity means 10 is missing in your food list so you should display the apple food with the price of 10 quantities in the general shopping list). Currently, You're showing all food in the general shopping list, and if any food is added to recipes you're removing it from the shopping list. Please make sure to fix this issue.

File: spec/integration/food_spec.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  it 'should edit and show a food' do
+    # create a food
+    visit foods_path
+    click_link 'New Food'
+    fill_in 'Name of food', with: 'test food'
+    fill_in 'Measurement unit', with: 'test unit'
+    fill_in 'Price', with: '1.0'
+    fill_in 'Quantity', with: '1'
+    click_button 'Create Food'
```
Comment (thecodechaser) [inline spec/integration/food_spec.rb:40]: - [ ] These two tests are failing, Please make sure to fix this and all tests should pass without any errors.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @sylvester-wamaya, and @ClintonYounge👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Highly organized code. ✔️
- Nice UI design. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi TEAM 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the required changes from the previous reviewer. 💯 

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**